File: leetcode/167.cpp
/* two sum II - input array is sorted
 * two sum with input array sorted
 */

#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        map<int, int> m; 
        for(int i = 0; i < numbers.size(); i++) {
            int n = numbers[i];
            if(m.count(n)) {
                vector<int> ans;
                ans.emplace_back(m[n]);
                ans.emplace_back(i+1);
                return ans;
            }
            m[target - n] = i+1;
        }
    return {};
    }
};

File: leetcode/167.rb
# two sum II - input array is sorted
# two sum with sorted (ascending order) input array

# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
  numbers.each_with_index {|x, i|
    j = numbers.bsearch_index{ |v| target - x <=> v }
    if j != nil and i != j
      return [i+1, j+1]
    end
  }    
end
